### ISSUE TITLE
[Question]How to use nndet to classify ribfrac

### ISSUE THREAD

**Issue by BelieferQAQ** [MIC-DKFZ/nnDetection#101]
### :question: Question

I want to classify rib fractures detected by ribfrac data set. What should I do. I have converted the ribfrac data set according to the format of nnunet (that is, 0, 1, 2, 3, 4, 5 in the mask correspond to the background and 5 fracture types). What should I do next? Thank you very much

**Comment by BelieferQAQ** [comment]
> 

Thank you. I'm going to try 5-classification detection on the ribfrac dataset. In addition, please tell me how to avoid 50% cross verification, or only verify one of them (I don't need to train 5 times, it takes too much time)

**Comment by BelieferQAQ** [comment]
Thank you very much for your answer. Hello, I'm doing the 5-type detection and classification of rib fracture of ribfrac. I have three questions to ask you for help:
1. This is my JSON file. I don't know target_ Class "and test_labels are set correctly?
{
"name": "RibFracFG",
"task": "Task020FG_RibFrac",
"target_class": null,
"test_labels": false,
"labels": {
"0": "displaced_rib_fracture",
"1": "non_displaced_rib_fracture",
"2": "buckle_rib_fracture",
"3": "segmental_rib_fracture",
"4": "unidentified_rib_fracture"
},
"modalities": {
"0": "CT"
},
"dim": 3
}
2. During the training, I found that the verification set loss is a negative number (is it better for nndet's loss and nnunet's loss to be closer to - 1?), Is it normal that the map value is very low (possibly because it is difficult to classify the ribfrac dataset)?
Epoch 11: 100%|██████████████████████████████████████████████████████████████████████████████████████████| 2600/2600  [2:53:33<00:00,  4.00s/it, loss=0.0127, v_num=e268INFO Val loss reached: -0.11723████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/ 100 [01:39<00:00,  3.64it/s]
INFO  mAP@0.1 :0.5:0.05: 0.080   AP@0.1 : 0.100   AP@0.5 : 0.052
INFO Proxy FG Dice: 0.736
INFO This epoch took 99 s
Epoch 11: 100%|██████████████████████████████████████████████████████████████████████████████████████████| 2600/2600 [2:53:34<00:00,  4.00s/it, loss=0.0127, v_num=e268INFO Train loss reached: -0.07363
3. I want to visualize the segmentation effect, but I don't know which instruction can generate the seg.pkl file required by the seg2nii() function?
res = load_ pickle(prediction_dir / f"{cid}_seg.pkl")

**Comment by mibaumgartner** [comment]
Sorry, I didn't see the follow up.

1) Looks good to me. 
2) The loss depends on the used losses. The GIoU loss can be negative in nnDetection since me omit the 1-loss computation. 
I would expect the mAP value to be quite low since the classification was very difficult (I never tried it myself but remember the results from the challenge). 
3) Extracting the segmentation is not officially supported in nnDetection. It is possible to set an additional flag in the inference script but that is highly experimental and thus I wouldn't recommend it. (use overwrite `do_seg=True`)
